lib/SILOptimizer/Transforms/PerformanceInliner.cpp

```diff
@@ -586,6 +608,10 @@ bool SILPerformanceInliner::isProfitableToInline(
         if (isa<AllocationInst>(retVal) || isa<PartialApplyInst>(retVal)) {
           BlockW.updateBenefit(Benefit, RemovedCallBenefit + 10);
           returnsAllocation = true;
+        } else if (auto n = isTupleWithAllocsOrPartialApplies(retVal); n > 0) {
+          // TODO: Do we need to tweak inlining benefits given to VJPs?
+          BlockW.updateBenefit(Benefit, n * (RemovedCallBenefit + 10));
```

Comment:
`n *` is not the right way. Benefits are not added but max'd. See https://github.com/apple/swift/blob/a94d22e017bd6a4078caec456de6b1b4719af248/lib/SILOptimizer/Utils/PerformanceInlinerUtils.cpp#L557
So it's sufficient to check if there is any such tuple element and update the benefit with `RemovedCallBenefit + 10` in such a case

Reply:
Changed. Thanks for taking a look!